src/test/java/picard/illumina/IlluminaBasecallsToSamTest.java

```diff
@@ -252,6 +254,54 @@ public void testDualBarcodes() throws Exception {
         runStandardTest(1, "dualBarcode.", "barcode_double.params", 1, "25T8B8B25T", DUAL_BASECALLS_DIR, DUAL_TEST_DATA_DIR);
     }
 
+    @Test
+    public void testCbclConvert() throws Exception {
+        runNewConverterTest(1, "dualBarcode.", "barcode_double.params", 2, "151T8B8B151T", TEST_DATA_DIR_WITH_CBCLS, DUAL_CBCL_TEST_DATA_DIR);
+    }
+
+    private void runNewConverterTest(final int lane, final String jobName, final String libraryParamsFile,
+                                     final int concatNColumnFields, final String readStructure, final File baseCallsDir,
+                                     final File testDataDir) throws Exception {
+        final File outputDir = File.createTempFile(jobName, ".dir");
+        outputDir.delete();
+        outputDir.mkdir();
+        outputDir.deleteOnExit();
```

Comment:
This kind of code for tests makes me wish we were using JUnit instead of TestNG. You could get rid of a lot of this cruft by using JUnit's `TemporaryFolder` rule.